core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -352,7 +352,7 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
                                     parentType, scoreMode, parentChildIndexFieldData, context.getSearchSimilarity());
     }
 
-    static final class LateParsingQuery extends Query {
+    public static final class LateParsingQuery extends Query {
```

Comment:
Especially now that this is public, can it please have some javadoc explaining what it does at all?